Assets/Scripts/Builder/Builder.cs

```diff
@@ -53,17 +55,26 @@ private void UpdateCurrentBuilding()
                 intersectPoint + Vector3.up * preViewOffset,
                 Quaternion.identity);
             currentBuildHelper = currentBuilding.GetComponent<BuildHelper>();
+
+            isCurrentBuildResourceExtractor = currentBuilding.TryGetComponent(out ResourceExtractor _);
         }
     }
 
     private void AddCurrentBuildingToMap()
     {
         if (currentBuilding)
         {
-            Instantiate(
-                currentBuildingPrefab,
-                currentBuilding.transform.position - Vector3.up * preViewOffset,
-                Quaternion.identity);
+            Vector3 position = currentBuilding.transform.position - Vector3.up * preViewOffset;
+            GameObject building = Instantiate(currentBuildingPrefab, position, Quaternion.identity);
+
+            if (isCurrentBuildResourceExtractor)
+            {
+                ResourceDeposit resourceDeposit = worldGenerator.GetResourceDeposit(position);
+                Debug.Assert(resourceDeposit, "Resource Deposit is null");
```

Comment:
Why is this should be null?

Reply:
If hex cell on this position doesn't contain resource, ```worldGenerator.GetResourceDeposit``` function will return null.

Reply:
But it asserts is should be null all the time

Reply:
Why?